src/hash.rs

```diff
@@ -64,3 +70,65 @@ fn file_hash(algorithm: &str, path: &str) -> Result<String> {
 
     Ok(hash_algorithm(algorithm, &bytes)?)
 }
+
+fn totp_generate(hex_seed: &str) -> Result<String> {
+    let mut seed: [u8; 64] = [0; 64];
+
+    let hex_seed_bytes = (0..hex_seed.len())
+                        .step_by(2)
+                        .map(|i| u8::from_str_radix(&hex_seed[i..i + 2], 16));
+
+    let mut i = 0;
+    for b in hex_seed_bytes {
+        seed[i] = b.unwrap();
+        i += 1;
+    }
```

Comment:
We already use the [hex](https://crates.io/crates/hex) crate in this module which could decode this input directly into to your array with `hex::decode_to_slice`.